### DISCUSSION TITLE
Custom Views (for ListOperation) PRO

### DISCUSSION THREAD

**Discussion by 4lbane** [Laravel-Backpack/community-forum#903] [Q&A (Help)]
Hello ! 

I think that I'm missing something about the new custom views for ListOperation (PRO)
I have a list of "Hot Air Balloons" and I need to display : 
- Hot Air Balloons that are not archived 
- Hot Air Balloons that are archived 

I have done this with :

```php
protected function setupListOperation()
    { 
      $this->runCustomViews([
                  'setupArchivesView' => __('Archives'),
              ]);
              
        CRUD::addClause('where', 'archive', '=', '0');
    }
    
    public function setupArchivesView()
    {
        CRUD::addClause('where', 'archive', '=', '1');
    }
``` 

But, in my 'setupArchivesView', I have 0 results and when I dump($this->crud), I see that the addClause from the setupListOperation() is applied : the first where has been combined with the second where. 

How can I remove the addClause from the setupListOperation() in the setupArchivesView() ? 

Thanks 🙏🙏

**Comment by karandatwani92** [comment]
Hey @pxpm

He is trying to alter a `where` clause. He is trying to set:
-  `CRUD::addClause('where', 'archive', '=', '0');` - In Default view
- `CRUD::addClause('where', 'archive', '=', '1');` - In Custom view

Do you know a way to do it? What condition can help here?

**Comment by karandatwani92** [comment]
Hey @4lbane 

Finally, I found the solution. We need to remove the base `where` clauses from the query coming from the default list operation.  The `addClause()` adds it, not meant to alter it).
The following code works:
```php
public function setupArchivesView(){
 $this->crud->query->getQuery()->wheres = []; // clear where clauses
 $this->crud->query->getQuery()->bindings['where'] = []; // clear where value params
 $this->crud->addClause('where', 'archive', '=', '1');
}
```

**Reply by pxpm** [reply]
This is no different than doing:
```php
$this->crud->query = $this->crud->model->where('active', '=', '1'); 
```
